packages/react-router/lib/hooks.tsx

```diff
@@ -685,18 +688,75 @@ export function _renderMatches(
     );
   }
 
+  // If we're in a partial hydration mode, detect if we need to render down to
+  // a given Fallback while we load the rest of the hydration data
+  let renderFallback = false;
+  let fallbackIndex = -1;
+  if (dataRouterState && future && future.v7_partialHydration) {
+    for (let i = 0; i < renderedMatches.length; i++) {
+      let match = renderedMatches[i];
+      // Track the deepest fallback up until the first route without data
+      if (match.route.Fallback || match.route.fallbackElement) {
+        fallbackIndex = i;
+      }
+      if (
+        match.route.loader &&
+        match.route.id &&
+        dataRouterState.loaderData[match.route.id] === undefined &&
+        (!dataRouterState.errors ||
+          dataRouterState.errors[match.route.id] === undefined)
+      ) {
+        // TODO:
+        // TODO: Add unit tests for this error flow!
+        // TODO:
+
```

Comment:
Note to self :)

Reply:
Added in [4841e12](https://github.com/remix-run/react-router/pull/11033/commits/4841e12bf820e152da1b28f94b832d8728474272)